crates/sui-core/src/authority_active/execution_driver/mod.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_types::{base_types::TransactionDigest, error::SuiResult, messages::CertifiedTransaction};
+use tracing::debug;
+use typed_store::Map;
+
+use crate::{authority::AuthorityState, authority_client::AuthorityAPI};
+
+use super::{gossip::LocalConfirmationTransactionHandler, ActiveAuthority};
+
+#[cfg(test)]
+pub(crate) mod tests;
+
+pub trait PendCertificateForExecution {
+    fn pending_execution(
+        &self,
+        certs: Vec<(TransactionDigest, CertifiedTransaction)>,
+    ) -> SuiResult<()>;
+}
+
+impl PendCertificateForExecution for AuthorityState {
+    fn pending_execution(
+        &self,
+        certs: Vec<(TransactionDigest, CertifiedTransaction)>,
+    ) -> SuiResult<()> {
+        self.database.add_pending_certificates(certs)
+    }
+}
+
+pub struct PendCertificateForExecutionNoop;
```

Comment:
Can we add a comment to explain when this is used?

Reply:
Yes, I added:
```
/// A no-op PendCertificateForExecution that we use for testing, when
/// we do not care about certificates actually being executed.
```